src/importer/sources/retroarch_source.py

```diff
@@ -0,0 +1,119 @@
+# retroarch_source.py
+#
+# Copyright 2023 Rilic
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from pathlib import Path
+from time import time
+
+import json
+import os
+import logging
+from json import JSONDecodeError
+
+from src import shared
+from src.game import Game
+from src.importer.sources.location import Location
+from src.importer.sources.source import (
+    SourceIterationResult,
+    SourceIterator,
+    Source
+)
+
+
+class RetroarchSourceIterator(SourceIterator):
+    source: "RetroarchSource"
+
+    def generator_builder(self) -> SourceIterationResult:
+        playlist_files = []
+        for file in os.listdir(self.source.config_location["playlists"]):
+            if file.endswith('.lpl'):
+                playlist_files.append(file)
+
+        playlist_items = []
+        for playlist_file in playlist_files:
+            open_file = open(str(self.source.config_location["playlists"]) + "/" + playlist_file)
+            try:
+                playlist_json = json.load(open_file)
+            except (JSONDecodeError, OSError, KeyError):
+                logging.warning("Cannot read playlist file: %s", str(playlist_file))
+                continue
+
+            for item in playlist_json["items"]:
+                # Select the core. Try the content's core first, then the playlist's
+                # default core.
+                core_path = item["core_path"]
+                if core_path == "DETECT":
+                    default_core = playlist_json["default_core_path"]
+                    if default_core:
+                        core_path = default_core
+                    else:
+                        logging.warning("Cannot find core for: %s", str(item["path"]))
+                        continue
+
+                # Build game
+                game_title = item["label"].split("(", 1)[0]
```

Comment:
Why not use the label itself?

Reply:
Labels in retroarch usually have extra suffixes appended to them, such as regions and revision numbers. I personally thought it looked a little bit ugly and didn't fit with the rest of the games shown in Cartridges, but on second thought that should be configurable through Retroarch itself, right?

Reply:
@GeoffreyCoulaud Actually, testing this reveals a bug with SteamGridDB covers - because it has the rest of those suffixes on the end when searching the game, it sometimes won't find a cover. These suffixes are default RetroArch behaviour, too.
Should I keep it as it is to keep it simple, revert back to truncating, or is there an alternative way of supplying the game name to SteamGridDB separate from the one shown on the UI?

Reply:
I fixed the problem I found, it was just a brainfart - I thought item["label"] was the filename for some reason :facepalm:

Reply:
I'm glad that works then :)
You can get rid of `game_title` and use the label directly then.